src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala

```diff
@@ -153,10 +152,21 @@ object DatabaseConversions extends LazyLogging {
 
         val untouchedAccounts =
           inactiveBakers
-            .map(_.copy(isActiveBaker = Some(false)))
-            .filterNot(accountsHistoryRow => touched.contains(accountsHistoryRow.accountId))
+            .filterNot(inactiveAccountsRow => touched.contains(inactiveAccountsRow.accountId))
+            .map {
+              _.into[Tables.AccountsHistoryRow]
+                .withFieldConst(
+                  _.asof,
+                  Timestamp.from(blockTaggedAccounts.timestamp.getOrElse(Instant.ofEpochMilli(0)))
+                )
+                .withFieldConst(_.cycle, blockTaggedAccounts.cycle)
+                .withFieldConst(_.blockLevel, blockTaggedAccounts.blockLevel)
+                .withFieldConst(_.blockId, blockTaggedAccounts.blockHash.value)
+                .withFieldConst(_.isActiveBaker, Some(false))
+                .transform
+            }
 
-        untouchedAccounts ::: touchedAccounts
+        (untouchedAccounts ::: touchedAccounts).distinct
```

Comment:
Question: if untouched already filtered out those accounts whose `id` is in `touched`, isn't this last step redundant?
Do you think it'll make things safer?

Reply:
just to make things safer